src/modules/sitemap/services/sitemap.service.ts

```diff
@@ -0,0 +1,205 @@
+import { Readable } from 'stream';
+
+import { ContentPagesService, DataService } from '@meemoo/admin-core-api';
+import { Injectable, InternalServerErrorException } from '@nestjs/common';
+import _ from 'lodash';
+import moment from 'moment';
+
+import { SITEMAP_XML_OBJECTS_SIZE } from '../sitemap.consts';
+import { SitemapConfig, SitemapItemInfo } from '../sitemap.types';
+
+import { VisitorSpaceStatus } from '~generated/database-aliases';
+import {
+	GetSitemapConfigDocument,
+	GetSitemapConfigQuery,
+} from '~generated/graphql-db-types-hetarchief';
+import { AssetsService } from '~modules/assets/services/assets.service';
+import { AssetFileType } from '~modules/assets/types';
+import { IeObjectLicense } from '~modules/ie-objects/ie-objects.types';
+import { IeObjectsService } from '~modules/ie-objects/services/ie-objects.service';
+import { SpacesService } from '~modules/spaces/services/spaces.service';
+
+@Injectable()
+export class SitemapService {
+	constructor(
+		protected dataService: DataService,
+		protected spacesService: SpacesService,
+		protected contentPagesService: ContentPagesService,
+		protected ieObjectsService: IeObjectsService,
+		protected assetsService: AssetsService
+	) {}
+
+	public async generateSitemap() {
+		const sitemapConfig = await this.getSitemapConfig();
+		const contentPagesPaths = await this.getContentPagesPaths();
+
+		const staticPages = ['/', '/bezoek', '/zoeken', '/geheime-content-pagina']; // /geheime-content-pagina is for testing
```

Comment:
we should remove /geheime-content-pagina since we can check blacklisting in the tests
if needed you could also do:

```
public async generateSitemap(staticPages: string[]) {
```

so you can pass the value from the controller and from the test